src/values/AbstractValue.js

```diff
@@ -510,7 +492,25 @@ export default class AbstractValue extends Value {
     left: void | Value,
     right: void | Value,
     loc?: ?BabelNodeSourceLocation
-  ): AbstractValue {
+  ): Value {
+    // c ? x : x <=> x
+    if (left !== undefined && right !== undefined && left.equals(right)) return left;
+    // x ? x : y <=> x || y
+    if (left instanceof AbstractValue && condition.equals(left) && right)
+      return AbstractValue.createFromLogicalOp(realm, "||", left, right, loc);
+    // y ? x : y <=> y && x
+    if (right instanceof AbstractValue && condition.equals(right) && left)
+      return AbstractValue.createFromLogicalOp(realm, "&&", right, left, loc);
+    // c ? (c ? x : y) : z <=> c ? x : z
```

Comment:
I am a bit vague on this. Is it safe to do this optimization while "y" has some side-effect? For example:
```js
var y = sideEffect();
var result = c ? (c ? x : y) : z;
```

Reply:
Arguments to abstract values must always be pure, so in that sense, it's safe.